kube-deploy/roles/kube-addons/tasks/addons-dashboard.yml

```diff
@@ -14,7 +14,18 @@
 # limitations under the License.
 #
 
+#For ARM support only
+- name: Checking for ARM support
+  shell: uname -a | grep -i "arm"
+  register: arm_enabled
+  ignore_errors: true
+
 - name: deploying the kubernetes dashboard
   shell: kubectl create -f https://rawgit.com/kubernetes/dashboard/master/src/deploy/kubernetes-dashboard.yaml
-  when: kube_addons_enabled and "{{kube_addons.kube_dashboard is defined}}"
+  when: kube_addons_enabled and "{{kube_addons.kube_dashboard is defined}}" and arm_enabled.rc != 0
+  ignore_errors: true
+
+- name: deploying the kubernetes dashboard - ARM support only
+  shell: curl -sSL https://rawgit.com/kubernetes/dashboard/master/src/deploy/kubernetes-dashboard.yaml | sed "s/amd64/arm64/g" | kubectl create -f -
```

Comment:
So if I am understanding your comments properly, if the `uname -m` output returns: `aarch64`, or `arm*`, then we need to replace the amd64 strings in the kubernetes manifest with `arm64`, correct? 

I am asking because I was thinking through an approach in which I grab the value of "ansible_architecture" (same output as uname -m) and using that value in your sed clause.. However, based on your examples, I don't think that is a valid approach.  I am trying to think through this from the perspective of supporting not only ARM and x86 but others as well.  Do you know if any other architectures are supported in Flannel, or is it just x86/64 and ARM?  If not, I will only need to test if "x86" is in the `ansible_architecture` variable and if it isn't, then use your sed clause to replace with arm. 

Let me know if this sounds like a good approach to move forward with.